packages/commerce-sdk-react/src/hooks/ShopperBaskets/cache.ts

```diff
@@ -51,9 +51,13 @@ const updateCustomerBasketsQuery = (
     return {
         queryKey: getCustomerBaskets.queryKey({...parameters, customerId}),
         updater: (oldData: CustomerBasketsResult | undefined) => {
-            // do not update if response basket is not part of existing customer baskets
+            // For createBasket, response basket is not part of existing customer baskets
+            // so we create a new result with 1 basket
             if (!oldData?.baskets?.some((basket) => basket.basketId === parameters.basketId)) {
-                return undefined
+                return {
+                    baskets: [response],
+                    total: 1
+                }
```

Comment:
I think we only will enter this scenario for createBasket so this should be fine as we are going from 0 to 1 basket

If not, this may allow us to update old entries with basket set [a,b,c] with a new entry with just basket set [d]

Reply:
This helper method is used by a bunch of endpoints. Rather than having a conditional in it that's just used by one endpoint, would it make sense to just create a different helper for that endpoint?

Reply:
With the change to the conditional to only run this if `oldData?.baskets?.length` is falsey, I think I think it makes sense now to leave this here and update the cache with 1 basket if there were no baskets in there prior regardless of which endpoint triggered the update.